presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -206,6 +207,9 @@ public static OrcPageSource createOrcPageSource(
             if (e.getClass().getSimpleName().equals("BlockMissingException")) {
                 throw new PrestoException(HIVE_MISSING_DATA, message, e);
             }
+            else if (e.getClass().getSimpleName().equals("InvalidProtocolBufferException")) {
```

Comment:
It's converted to a Presto exception, but with `HIVE_CANNOT_OPEN_SPLIT` error code. This change marks it with the `HIVE_BAD_DATA` error code.

```
	com.facebook.presto.spi.PrestoException: Error opening Hive split XXX): Malformed ORC file. Invalid compressed stream [XXX]
	at com.facebook.presto.hive.orc.OrcPageSourceFactory.createOrcPageSource(OrcPageSourceFactory.java:208)
	at com.facebook.presto.hive.orc.DwrfPageSourceFactory.createPageSource(DwrfPageSourceFactory.java:73)
	at com.facebook.presto.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:132)
	at com.facebook.presto.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:87)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:44)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:56)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:219)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:303)
	at com.facebook.presto.operator.Driver.lambda$processFor$6(Driver.java:234)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:537)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:229)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:623)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:463)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.facebook.presto.orc.protobuf.InvalidProtocolBufferException: Malformed ORC file. Invalid compressed stream [XXX]
	at com.facebook.presto.orc.proto.DwrfProto$Footer.<init>(DwrfProto.java:9899)
	at com.facebook.presto.orc.proto.DwrfProto$Footer.<init>(DwrfProto.java:9788)
	at com.facebook.presto.orc.proto.DwrfProto$Footer$1.parsePartialFrom(DwrfProto.java:9935)
	at com.facebook.presto.orc.proto.DwrfProto$Footer$1.parsePartialFrom(DwrfProto.java:9930)
	at com.facebook.presto.orc.protobuf.AbstractParser.parseFrom(AbstractParser.java:89)
	at com.facebook.presto.orc.protobuf.AbstractParser.parseFrom(AbstractParser.java:95)
	at com.facebook.presto.orc.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)
	at com.facebook.presto.orc.proto.DwrfProto$Footer.parseFrom(DwrfProto.java:10332)
	at com.facebook.presto.orc.metadata.DwrfMetadataReader.readFooter(DwrfMetadataReader.java:82)
	at com.facebook.presto.orc.OrcReader.<init>(OrcReader.java:159)
	at com.facebook.presto.hive.orc.OrcPageSourceFactory.createOrcPageSource(OrcPageSourceFactory.java:165)
	... 16 more
```